src/compiler/checker.ts

```diff
@@ -17584,15 +17540,16 @@ namespace ts {
             return getJsxElementTypeAt(node) || anyType;
         }
 
-        function checkJsxFragment(node: JsxFragment, checkMode: CheckMode | undefined): Type {
-            checkJsxOpeningLikeElementOrOpeningFragment(node.openingFragment, checkMode);
+        function checkJsxFragment(node: JsxFragment): Type {
+            checkJsxOpeningLikeElementOrOpeningFragment(node.openingFragment);
 
             if (compilerOptions.jsx === JsxEmit.React && (compilerOptions.jsxFactory || getSourceFileOfNode(node).pragmas.has("jsx"))) {
                 error(node, compilerOptions.jsxFactory
                     ? Diagnostics.JSX_fragment_is_not_supported_when_using_jsxFactory
                     : Diagnostics.JSX_fragment_is_not_supported_when_using_an_inline_JSX_factory_pragma);
             }
 
+            checkJsxChildren(node);

```

Comment:
so, uhh, we weren't checking children before?

Reply:
If the attributes type is non-any, they can get checked via their contributions to the attributes type; _however_ we need to ensure they get visited even in the any/error cases, so we do that here.